strings/base_error.h

```diff
@@ -539,24 +539,11 @@ namespace winrt::impl
         return result;
     }
 
-    inline WINRT_IMPL_NOINLINE void check_cast_result(void* from, winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current())
+    inline WINRT_IMPL_NOINLINE void check_cast_result(hresult const result, winrt::impl::slim_source_location const& sourceInformation = winrt::impl::slim_source_location::current())
     {
-        if (!from)
+        if (result != 0)
```

Comment:
The distinction is that `check_cast_result` is tagged `noinline` so there isn't a bunch of new exception throwing code when `winrt::impl::consume_X` is inlined into callers.

Reply:
> The distinction is that `check_cast_result` is tagged `noinline` so there isn't a bunch of new exception throwing code when `winrt::impl::consume_X` is inlined into callers.

`throw_hresult` is also tagged `noinline`, so the only code that could get inlined into `consume_X` is `check_hresult` testing the value against 0.

I guess the other distinction is that `check_hresult` does `if (result < 0)`, while this function is doing `if (result != 0)`. So `check_cast_result` is going to originate errors for stuff like `S_FALSE`, which may not be a behavior we want to introduce. Just another argument for reusing rather than introducing a new function.

Reply:
Whoops, I'm sorry.  I looked at `check_hresult` not `throw_hresult`.  You're right.